src/Altinn.App.Core/Features/Validation/ExpressionValidator.cs

```diff
@@ -0,0 +1,275 @@
+        {
+            var rawValidationConfig = appResourceService.GetValidationConfiguration(dataType);
+            if (rawValidationConfig == null)
+            {
+                // No validation configuration exists for this data type
+                return new List<ValidationIssue>();
+            }
+
+            var validationConfig = JsonDocument.Parse(rawValidationConfig).RootElement;
+            return Validate(validationConfig, dataModel, evaluatorState, logger);
+        }
+
+        /// <inheritdoc />
+        public static IEnumerable<ValidationIssue> Validate(JsonElement validationConfig, IDataModelAccessor dataModel, LayoutEvaluatorState evaluatorState, ILogger logger)
+        {
+            var validationIssues = new List<ValidationIssue>();
+            var expressionValidations = ParseExpressionValidationConfig(validationConfig, logger);
+            foreach (var validationObject in expressionValidations)
+            {
+                var baseField = validationObject.Key;
+                var resolvedFields = dataModel.GetResolvedKeys(baseField);
+                var validations = validationObject.Value;
+                foreach (var resolvedField in resolvedFields)
+                {
+                    var positionalArguments = new[] { resolvedField };
+                    foreach (var validation in validations)
+                    {
+                        try
+                        {
+                            if (validation.Condition == null)
+                            {
+                                continue;
+                            }
+
+                            var isInvalid = ExpressionEvaluator.EvaluateExpression(evaluatorState, validation.Condition, null, positionalArguments);
+                            if (isInvalid is not bool)
+                            {
+                                throw new ArgumentException($"Validation condition for {resolvedField} did not evaluate to a boolean");
+                            }
+                            if ((bool)isInvalid)
+                            {
+                                var validationIssue = new ValidationIssue
+                                {
+                                    Field = resolvedField,
+                                    Severity = validation.Severity ?? ValidationIssueSeverity.Error,
+                                    CustomTextKey = validation.Message,
+                                    Code = validation.Message,
+                                    Source = "Expression" // TODO: Add source to ValidationIssueSources
+                                };
+                                validationIssues.Add(validationIssue);
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            logger.LogError($"Error while evaluating expression validation for {resolvedField}: {e}");
+                        }
```

Comment:
@tjololo is it bad practice to catch errors when evaluating expression validations? Should I rather let it throw an exception and fail? I suppose you could circumvent a validation if you could make it crash.

Reply:
You'd definitely circumvent any validation if you just catch-log-continue.

I would hope an exception in validation would prevent users from submitting anything?

Reply:
We should definitely not continue as nothing happened.
Either we just let the error be thrown to the caller, this would in the en probably lead to an 500 response that frontend needs to handle. Another possibility is to add a validation issue when an exception occurs. The error could then indicate that there was a problem while validation the data.